zebra-consensus/src/block/subsidy.rs

```diff
@@ -0,0 +1,216 @@
+        panic!("can't verify before block {}", SLOW_START_INTERVAL.0)
+    }
+    let blossom_height = Blossom
+        .activation_height(network)
+        .expect("blossom activation height should be available");
+
+    let hd = halving_divisor(height, network);
+    if height >= blossom_height {
+        let scaled_max_block_subsidy = MAX_BLOCK_SUBSIDY / BLOSSOM_POW_TARGET_SPACING_RATIO;
+        // in future halvings, this calculation might not be exact
+        // in those cases, Amount division follows integer division, which truncates (rounds down) the result
+        Amount::try_from(scaled_max_block_subsidy / hd)
+    } else {
+        // this calculation is exact, because the halving divisor is 1 here
+        Amount::try_from(MAX_BLOCK_SUBSIDY / hd)
+    }
+}
+
+/// `FoundersReward(height)` as described in [protocol specification §7.7][7.7]
+///
+/// [7.7]: https://zips.z.cash/protocol/protocol.pdf#subsidies
+pub fn founders_reward(height: Height, network: Network) -> Result<Amount<NonNegative>, Error> {
+    if halving_divisor(height, network) == 1 {
+        // this calculation is exact, because the block subsidy is divisible by
+        // the FOUNDERS_FRACTION_DIVISOR until long after the first halving
+        block_subsidy(height, network)? / FOUNDERS_FRACTION_DIVISOR
+    } else {
+        Amount::try_from(0)
+    }
+}
+/// `MinerSubsidy(height)` as described in [protocol specification §7.7][7.7]
+///
+/// [7.7]: https://zips.z.cash/protocol/protocol.pdf#subsidies
+pub fn miner_subsidy(height: Height, network: Network) -> Result<Amount<NonNegative>, Error> {
+    let canopy_height = Canopy
+        .activation_height(network)
+        .expect("canopy activation height should be available");
+    if height >= canopy_height {
+        panic!("Can't validate Canopy yet");
+    }
+    block_subsidy(height, network)? - founders_reward(height, network)?
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use color_eyre::Report;
+
+    #[test]
+    fn test_halving() -> Result<(), Report> {
+        let network = Network::Mainnet;
+        let blossom_height = Blossom.activation_height(network).unwrap();
+        let canopy_height = Canopy.activation_height(network).unwrap();
+
+        assert_eq!(1, halving_divisor((blossom_height - 1).unwrap(), network));
+        assert_eq!(1, halving_divisor(blossom_height, network));
+        assert_eq!(1, halving_divisor((canopy_height - 1).unwrap(), network));
+        assert_eq!(2, halving_divisor(canopy_height, network));
+        assert_eq!(2, halving_divisor((canopy_height + 1).unwrap(), network));
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_block_subsidy() -> Result<(), Report> {
+        let network = Network::Mainnet;
+        let blossom_height = Blossom.activation_height(network).unwrap();
+        let canopy_height = Canopy.activation_height(network).unwrap();
+
+        // Before the 1st halving(at the same time of Canopy) the block reward is 6.25 ZEC
```

Comment:
This numbers are not specified in the protocol, we can maybe add a reference to https://z.cash/support/faq/#mining instead. It is not possible to link to the exact section we want which is **How many ZEC are made per block?**

Reply:
Sounds good

Reply:
It turns out that i can link to the actual question instead of the section in the FAQ, applied this at https://github.com/ZcashFoundation/zebra/pull/1051/commits/1a890d25c7b622d3d058f3cce63b84b6d7809bad